packages/server/customer-os-webhooks/service/sync_email_service.go

```diff
@@ -35,47 +44,267 @@ func NewSyncEmailService(log logger.Logger, repositories *repository.Repositorie
 	}
 }
 
-func (s syncEmailService) SyncEmail(ctx context.Context, email model.EmailData) (SyncResult, error) {
-	var organizationsData []model.OrganizationData
-	var domainsSlice []string
+func (s syncEmailService) SyncEmail(ctx context.Context, emailData model.EmailData) (organizationSync SyncResult, interactionEventSync SyncResult, contactSync SyncResult, err error) {
 	var name string
+	var orgSyncResult, interactionEventSyncResult, contactSyncResult SyncResult
 
 	span, ctx := opentracing.StartSpanFromContext(ctx, "SyncEmailService.SyncEmails")
 	defer span.Finish()
 	tracing.SetDefaultServiceSpanTags(ctx, span)
-	tracing.LogObjectAsJson(span, "email", email)
+	tracing.LogObjectAsJson(span, "emailData", emailData)
 
 	if !s.services.TenantService.Exists(ctx, common.GetTenantFromContext(ctx)) {
 		s.log.Errorf("tenant {%s} does not exist", common.GetTenantFromContext(ctx))
 		tracing.TraceErr(span, errors.ErrTenantNotValid)
-		return SyncResult{}, errors.ErrTenantNotValid
+		return SyncResult{}, SyncResult{}, SyncResult{}, errors.ErrTenantNotValid
 	}
 
-	domain := ""
-	if email.SentBy != "" {
-		domain = utils.ExtractDomain(email.SentBy)
-		utils.EnforceSingleValue(domainsSlice, domain)
-		name = utils.ExtractName(email.SentBy)
+	if strings.HasSuffix(emailData.Subject, "• lemwarmup") || strings.HasSuffix(emailData.Subject, "• lemwarm") {
+		return SyncResult{Skipped: 1}, SyncResult{}, SyncResult{}, nil
 	}
 
-	// TODO if email present in COS, skip org creation, contact creation, email node creation, same for to / cc / bcc
-	// TODO for each email -> create org, contact email node. -> Check Edi logic
-	organizationData := model.OrganizationData{
-		BaseData: model.BaseData{
-			AppSource: email.AppSource,
-			Source:    email.ExternalSystem,
+	interactionEventId, err := s.repositories.InteractionEventRepository.GetInteractionEventIdByExternalId(ctx, common.GetTenantFromContext(ctx), emailData.ExternalId, emailData.ExternalSystem)
+	if err != nil {
+		logrus.Errorf("failed to check if interaction event exists for external id %v for tenant %v :%v", emailData.ExternalId, common.GetTenantFromContext(ctx), err)
+		return SyncResult{}, SyncResult{Failed: 1}, SyncResult{}, nil
+	}
+
+	if interactionEventId == "" {
+
+		now := time.Now().UTC()
+
+		emailSentDate, err := s.ConvertToUTC(emailData.CreatedAtStr)
+		if err != nil {
+			logrus.Errorf("failed to convert emailData sent date to UTC for emailData with id %v :%v", emailData.Id, err)
+		}
+
+		from, to, cc, bcc, references, inReplyTo := extractEmailData(emailData)
+
+		personalEmailProviderList, err := s.services.CommonServices.CommonRepositories.PersonalEmailProviderRepository.GetPersonalEmailProviders()
```

Comment:
this needs to be moved way upper. this is a 7k list in production. we need to fetch it on container startup and propagate it down the stream